pkg/kafka_ingester/kafka_ingester_test.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/seznam/slo-exporter/pkg/stringmap"
+)
+
+func Test_processMessage(t *testing.T) {
+	tests := []struct {
+		Name         string
+		KafkaMessage kafka.Message
+		OutputEvent  *event.Raw
+		ErrExpected  bool
+	}{
+		{
+			Name: "Empty event",
+			KafkaMessage: kafka.Message{
+				Topic: "topic",
+				Value: []byte(`{}`),
+			},
+			OutputEvent: &event.Raw{Quantity: 1},
+		},
+		{
+			Name: "Event without classification",
+			KafkaMessage: kafka.Message{
+				Topic: "topic",
+				Value: []byte(`{
+					"quantity": 1,
+					"metadata": {"foo": "bar"}
+				}`),
+			},
+			OutputEvent: &event.Raw{Quantity: 1, Metadata: stringmap.StringMap{"foo": "bar"}},
+		},
+		{
+			Name: "Event with classification",
+			KafkaMessage: kafka.Message{
+				Topic: "topic",
+				Value: []byte(`{
+					"quantity": 1,
+					"metadata": {"foo": "bar"},
+					"slo_classification": {"app": "fooApp", "class": "fooClass", "domain": "fooDomain"}
+				}`),
+			},
+			OutputEvent: &event.Raw{
+				Quantity:          1,
+				Metadata:          stringmap.StringMap{"foo": "bar"},
+				SloClassification: &event.SloClassification{App: "fooApp", Class: "fooClass", Domain: "fooDomain"},
+			},
+		},
+		{
+			Name: "Kafka message containing only unknown fields",
+			KafkaMessage: kafka.Message{
+				Topic: "topic",
+				Value: []byte(`{"unknown_field": 1111}`),
+			},
+			OutputEvent: &event.Raw{Quantity: 1},
+		},
+		{
+			Name: "Valid event data accompanied with unknown fields",
+			KafkaMessage: kafka.Message{
+				Topic: "topic",
+				Value: []byte(`{
+					"quantity": 1,
+					"metadata": {"foo": "bar"},
+					"slo_classification": {"app": "fooApp", "class": "fooClass", "domain": "fooDomain"},
+					"unknown_fields": "foo"
+				}`)},
+			OutputEvent: &event.Raw{
+				Quantity:          1,
+				Metadata:          stringmap.StringMap{"foo": "bar"},
+				SloClassification: &event.SloClassification{App: "fooApp", Class: "fooClass", Domain: "fooDomain"},
+			},
+		},
+		{
+			Name: "Valid event data, explicit schema version",
+			KafkaMessage: kafka.Message{
+				Headers: []kafka.Header{{schemaVersionMessageHeader, []byte("v1")}},
+				Topic:   "topic",
+				Value: []byte(`{
+					"quantity": 1,
+					"metadata": {"foo": "bar"},
+					"slo_classification": {"app": "fooApp", "class": "fooClass", "domain": "fooDomain"}
+				}`)},
+			OutputEvent: &event.Raw{
+				Quantity:          1,
+				Metadata:          stringmap.StringMap{"foo": "bar"},
+				SloClassification: &event.SloClassification{App: "fooApp", Class: "fooClass", Domain: "fooDomain"},
+			},
+		},
+		{
+			Name: "Valid event data, unknown schema version",
+			KafkaMessage: kafka.Message{
+				Headers: []kafka.Header{{schemaVersionMessageHeader, []byte("unknown")}},
+				Topic:   "topic",
+				Value: []byte(`{
+					"quantity": 1,
+					"metadata": {"foo": "bar"},
+					"slo_classification": {"app": "fooApp", "class": "fooClass", "domain": "fooDomain"}
+				}`)},
+			OutputEvent: nil,
+			ErrExpected: true,
+		},
```

Comment:
nit: Maybe one more tests case with unsuccessful marshalling expecting error would be nice? For example invalid type or something like that?

Reply:
be7b68f